plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -712,6 +748,60 @@ private void finishOptimize(IcebergTableExecuteHandle executeHandle, Collection<
         transaction = null;
     }
 
+    @Override
+    public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        IcebergTableExecuteHandle executeHandle = (IcebergTableExecuteHandle) tableExecuteHandle;
+        switch (executeHandle.getProcedureId()) {
+            case VACUUM:
+                executeVacuum(session, executeHandle);
+                return;
+            default:
+                throw new IllegalArgumentException("Unknown procedure '" + executeHandle.getProcedureId() + "'");
+        }
+    }
+
+    private void executeVacuum(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        IcebergVacuumHandle vacuumHandle = (IcebergVacuumHandle) executeHandle.getProcedureHandle();
+        Table table = catalog.loadTable(session, executeHandle.getSchemaTableName());
+        long retentionThreshold = vacuumHandle.getRetentionThreshold().toMillis();
+        long expireTimestampMillis = System.currentTimeMillis() - retentionThreshold;
+        table.expireSnapshots()
+                .expireOlderThan(expireTimestampMillis)
+                .cleanExpiredFiles(true)
+                .commit();
+        deleteOrphanFiles(table, session, executeHandle.getSchemaTableName(), expireTimestampMillis);
```

Comment:
Does `cleanExpiredFiles` not delete everything?

Reply:
unfortunately it doesn't delete data files see https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1643737841905979

Reply:
`cleanExpiredFiles` is set to `true` in `org.apache.iceberg.RemoveSnapshots`

Reply:
@findinpath I prefer to set it explicitly to make sure it won't change between versions, it would lead silent bugs